src/NServiceBus.Core.Tests/AssemblyScanner/AssemblyScannerTests.cs

```diff
@@ -268,12 +310,30 @@ public DynamicAssembly(string nameWithoutExtension, DynamicAssembly[] references
                     param.ReferencedAssemblies.Add(reference.FilePath);
                 }
 
-                builder.AppendLine("public class Foo { public Foo() {");
-                foreach (var reference in references)
+                if (referenceTheCore)
```

Comment:
Not really @timbussmann The core assembly name is here to mimic scenarios where we only introduce things that reference the assembly with that name. We need this hack to also test multiple versions of fake core. The valid use case @andreasohlund and I introduced here is that we essentially allow referencing the latest real core